category_encoders/target_encoder.py

```diff
@@ -344,3 +311,43 @@ def _weighting(self, n: int) -> float:
         # monotonically increasing function of n bounded between 0 and 1
         # sigmoid in this case, using scipy.expit for numerical stability
         return expit((n - self.min_samples_leaf) / self.smoothing)
+
+    def _generate_inverted_hierarchy(self) -> tuple[dict | pd.DataFrame, dict]:
```

Comment:
this returns a `tuple[dict | pd.DataFrame | None, dict]` right? If self.hierarchy is None this will also be None.
I appreciate the type hints btw. I'm working on a branch to add typing support to the whole library.

Reply:
Good catch.  I made this change.